### ISSUE TITLE
bug: API GW @connections endpoint error

### ISSUE THREAD

**Issue by DaniEisenberg** [localstack/localstack#9959]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

1. Create API Gateway  websockets API deployment
2. Connect to Websockets and pass connectionId
3. use connectionId in order to pass data back to the client - error

Request: POST http://{apiId}.execute-api.localhost.localstack.cloud:4566/{stage-name}/@connections/{connectionId}
Response: 404     "error": "Unable to find integration for API /@connections/{connectionId} route ccc5dec4: "

### Expected Behavior

I expect that I can call @connections endpoint the same way I'm doing this on AWS (described here: [docs](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-how-to-call-websocket-api-connections.html))

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

apiId='wsapi'
awslocal apigatewayv2 create-api --name "myWebSocketApi3" --protocol-type WEBSOCKET --route-selection-expression '$request.body.action' --tags="_custom_id_=wsapi"

awslocal apigatewayv2 create-integration --api-id $apiId --integration-type=HTTP_PROXY --passthrough-behavior WHEN_NO_MATCH --timeout-in-millis=29000 --integration-method POST --request-parameters 'integration.request.header.ConnectionId'='context.connectionId' --integration-uri=http://host.docker.internal:8080/websockets/api/disconnect

awslocal apigatewayv2 create-integration --api-id $apiId --integration-type=HTTP_PROXY --passthrough-behavior WHEN_NO_MATCH --timeout-in-millis=29000 --integration-method POST --request-parameters 'integration.request.header.ConnectionId'='context.connectionId' --integration-uri=http://host.docker.internal:8080/websockets/api/connect

awslocal apigatewayv2 create-route --api-id $apiId --route-key '$disconnect' --no-api-key-required --authorization-type NONE --target $disconnectIntegration

awslocal apigatewayv2 create-route --api-id $apiId --route-key '$connect' --no-api-key-required --authorization-type NONE --target $connectIntegration

awslocal apigatewayv2 create-route --api-id $apiId --route-key '$default' --no-api-key-required --authorization-type NONE

awslocal apigatewayv2 create-deployment --api-id $apiId
awslocal apigatewayv2 create-stage --api-id $apiId --deployment-id $deploymentId --stage-name local-websockets

### Environment

```markdown
- OS: Sonoma 14
- LocalStack: latest
```

### Anything else?

The connectionId header passed correctly and saved on the backend. 
I'm using connectionId later on to make POST requests back to the API GW

I assume that url I'm using should be built like this: 

http://{apiId}.execute-api.localhost.localstack.cloud:4566/{stage-name}/@connections/{connectionId}

I'm using hardcoded apiId = wsapi so it always stay the same.

I have also added network alias with hardcoded api id to the compose: 
   ```
 networks:
      default:
        aliases:
          - localhost.localstack.cloud
          - wsapi.execute-api.localhost.localstack.cloud
```

It could be that url format is wrong OR I'm missing something while configuring the API GW. 
Any help would be much appreciated. 
Thanks.

**Comment by calvernaz** [comment]
Hi @DaniEisenberg thanks for reporting this. I have tested and it works as expected but for completeness' sake, I want to understand if the variables you are expecting are just the **connectionId** or stage and apiID?

I'm asking because while I don't have any issue with the request template for the connectionId, I found an issue with the stage name.

1. Run step by step. 
2. Created a backend endpoint using Nodejs which will be called the backend integration. Which also print the headers.
```
const http = require('http');

const server = http.createServer((req, res) => {
    console.log(req.headers);
    res.statusCode = 200;
    res.setHeader('Content-Type', 'text/plain');
    res.end('Hello World\n');
});

server.listen(8080, '127.0.0.1', () => {
    console.log('Server running at http://127.0.0.1:8080/');
});
``` 

3. Connected to the websocket using the following `wscat -c ws://localhost.localstack.cloud:4510`, that resulted in the following log on the nodejs server:
```
{
....
  connectionid: '455b3812',
  'content-length': '22'
}
```

So works just fine! But as I said I changed one of the steps to be fully complete.

1. The create integration step for the `connect` route
```
awslocal apigatewayv2 create-integration --api-id $apiId --integration-type=HTTP_PROXY --passthrough-behavior WHEN_NO_MATCH --timeout-in-millis=29000 --integration-method POST --request-parameters 'integration.request.header.ConnectionId=context.connectionId,integration.request.header.apiId=context.apiId,integration.request.header.stageName=context.stage' --integration-uri=http://localhost:8080/websockets/api/connect
```

2. Connect the websocket:
```
{
...
  connectionid: '455b3812',
  apiid: 'wsapi',
  stagename: '',
```

Stage came up empty, so I'm fixing that and should be ok.

**Comment by DaniEisenberg** [comment]
Hi @calvernaz ! Thanks for your reply. However the problem is different, i was not super clear about the issue. So you are right connecting to the WebSockets works perfectly. There is no problem with that. 
But could you try to send data back to the connected client using connectionId? 

The way I do it is as follows: 
Request: POST http://{apiId}.execute-api.localhost.localstack.cloud:4566/{stage-name}/@connections/{connectionId}

This is what I get: 
Response: 404 "error": "Unable to find integration for API /@connections/{connectionId} route ccc5dec4: "

So my question is not about connecting to websockets but sending data back to the connected client using @connections endpoint. 
Thanks!

**Comment by calvernaz** [comment]
Ah...gotcha! I'm on track to this one as well, I notice when not using the clients API like https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/apigatewaymanagementapi/client/post_to_connection.html or the equivalent JS version (or any language SDK), the request is not being routed properly.

So, I'm also tackling that one.

**Comment by DaniEisenberg** [comment]
To be honest I'm using Java SDK to make this call. In particular, I'm using: aws-java-sdk-apigatewaymanagementapi library of version: 1.12.571

I'm configuring the client this way: 
`AmazonApiGatewayManagementApiClientBuilder.standard()
        .withCredentials(awsCredentialsProvider)
        .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(websocketConnectionUrl, region))
        .build();`

where awsCredentialsProvider is the one used to point to localstack
websocketConnectionUrl is built like this: `http://{apiId}.execute-api.localhost.localstack.cloud:4566/{stage-name}/@connections/{connectionId}`

then I'm executing:

`amazonApiGatewayManagementApi.postToConnection()`

Which works fine on AWS. I'm personally not sure about the URL format for localstack. Thanks for your effort!

**Comment by calvernaz** [comment]
@DaniEisenberg that's right! The endpoint generated contains the `{stage-name}` and that is what's causing issues. If you don't care for a moment about the stage path, you can try to use the endpoint like this `http://localhost.localstack.cloud:4566/{stage-name}/@connections/{connectionId}`, it's not optimal but should unblock you for now. 

I'm still working on this, and trying to find a away that works with the correct endpoint without breaking the existing implementations.

**Comment by DaniEisenberg** [comment]
@calvernaz I'm sorry, did you mean apiId? Looks like you've confused apiId and stagename.

So to confirm. Instead of this url:
`http://{apiId}.localhost.localstack.cloud:4566/{stage-name}/@connections/{connectionId}`
I should use: 
`http://localhost.localstack.cloud:4566/{stage-name}/@connections/{connectionId}`

Thank you for your help!

**Comment by calvernaz** [comment]
Hi @DaniEisenberg the fix should be available in the latest image, so could you give it a go 👍
